darwin/item.py

```diff
@@ -19,10 +59,101 @@ class DatasetItem:
 
     @property
     def full_path(self) -> str:
+        """
+        The full POSIX relative path of this ``DatasetItem``.
+        """
         return construct_full_path(self.path, self.filename)
 
+    @classmethod
+    def parse(cls, raw: Dict[str, Any]) -> "DatasetItem":
+        """
+        Parses the given dictionary into a ``DatasetItem``. Raises if such is not possible.
+
+        Parameters
+        ----------
+        raw : Dict[str, Any]
+            The dictionary to parse.
+
+        Returns
+        -------
+        DatasetItem
+            A dataset item with the parsed information.
+
+        Raises
+        ------
+        ValueError
+            If any of the keys from the given dictionary do not have the correct format.
+        """
+        id: int = raw["id"]
+        if not isinstance(id, int):
+            raise ValueError("Key 'id' must have an integer for a value.")
+
+        filename: str = raw["filename"]
+        if not isinstance(filename, str):
+            raise ValueError("Key 'filename' must have a string for a value.")
+
+        status: str = raw["status"]
+        if not isinstance(status, str):
+            raise ValueError("Key 'status' must have a string for a value.")
+
+        archived: bool = raw["archived"]
+        if not isinstance(archived, bool):
+            raise ValueError("Key 'archived' must have a boolean for a value.")
 
+        file_size: int = raw["file_size"]
+        if not isinstance(file_size, int):
+            raise ValueError("Key 'file_size' must have an integer for a value.")
+
+        dataset_id: int = raw["dataset_id"]
+        if not isinstance(dataset_id, int):
+            raise ValueError("Key 'dataset_id' must have an integer for a value.")
+
+        sequence: int = raw["seq"]
+        if not isinstance(sequence, int):
+            raise ValueError("Key 'seq' must have an integer for a value.")
+
+        workflow_id: Optional[int] = raw.get("current_workflow_id")
+        if not isinstance(workflow_id, int) and workflow_id is not None:
+            raise ValueError("Key 'workflow_id' must have an integer for a value or be nil.")
+
+        path: str = raw["path"]
+        if not isinstance(path, str):
+            raise ValueError("Key 'path' must have a string for a value.")
+
+        return DatasetItem(
+            id,
+            filename,
+            status,
+            archived,
+            file_size,
+            dataset_id,
+            "n/a",
+            sequence,
+            workflow_id,
+            path,
+        )
+
+
+@deprecation.deprecated(
+    deprecated_in="0.7.5",
+    removed_in="0.7.7",
```

Comment:
Mmh, I think we should removing it in the next minor or major version number, not patch IMHO
```suggestion
    removed_in="0.8.0",
```

Reply:
Agree, that was a placeholder :D